src/main/java/com/azure/cosmos/examples/diagnostics/async/CosmosDiagnosticsQuickStartAsync.java

```diff
@@ -0,0 +1,284 @@
+
+
+        createDatabaseIfNotExists();
+        createContainerIfNotExists();
+
+        createDocument();
+        readDocumentById();
+        queryDocuments();
+        replaceDocument();
+        upsertDocument();
+    }
+
+    // Database Diagnostics
+    private void createDatabaseIfNotExists() throws Exception {
+        logger.info("Creating database " + databaseName + " if not exists...");
+
+        //  Create database if not exists
+        Mono<CosmosDatabaseResponse> databaseResponseMono = client.createDatabaseIfNotExists(databaseName);
+        CosmosDatabaseResponse cosmosDatabaseResponse = databaseResponseMono.flatMap(databaseResponse -> {
+            CosmosDiagnostics diagnostics = databaseResponse.getDiagnostics();
+            logger.info("Create database diagnostics : {}", diagnostics);
+            return Mono.just(databaseResponse);
+        }).block();
+
+        database = client.getDatabase(cosmosDatabaseResponse.getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Container create
+    private void createContainerIfNotExists() throws Exception {
+        logger.info("Creating container " + containerName + " if not exists.");
+
+        //  Create container if not exists
+        CosmosContainerProperties containerProperties =
+            new CosmosContainerProperties(containerName, "/lastName");
+
+        // Provision throughput
+        ThroughputProperties throughputProperties = ThroughputProperties.createManualThroughput(400);
+
+        //  Create container with 200 RU/s
+        Mono<CosmosContainerResponse> containerResponseMono = database.createContainerIfNotExists(containerProperties,
+            throughputProperties);
+        CosmosContainerResponse cosmosContainerResponse = containerResponseMono.flatMap(containerResponse -> {
+            CosmosDiagnostics diagnostics = containerResponse.getDiagnostics();
+            logger.info("Create container diagnostics : {}", diagnostics);
+            return Mono.just(containerResponse);
+        }).block();
+        container = database.getContainer(cosmosContainerResponse.getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    private void createDocument() throws Exception {
+        logger.info("Create document " + documentId);
+
+        // Define a document as a POJO (internally this
+        // is converted to JSON via custom serialization)
+        Family family = new Family();
+        family.setLastName(documentLastName);
+        family.setId(documentId);
+
+        // Insert this item as a document
+        // Explicitly specifying the /pk value improves performance.
+        Mono<CosmosItemResponse<Family>> itemResponseMono = container.createItem(family,
+            new PartitionKey(family.getLastName()),
+            new CosmosItemRequestOptions());
+
+        itemResponseMono.flatMap(itemResponse -> {
+            CosmosDiagnostics diagnostics = itemResponse.getDiagnostics();
```

Comment:
also you may want to capture the failure exception case. how do we log diagnostics in that scenario?

Reply:
Very good point, I will add those cases as well.

Reply:
Done.